specs/latest/2.0/index.html

```diff
@@ -3202,6 +3246,62 @@ <h2 class="no-toc">Table of contents</h2>
         <code>FLOAT_32_UNSIGNED_INT_24_8_REV</code>, generates an <code>INVALID_ENUM</code>.
     </p>
 
+    <h3><a name="ATTRIBUTE_TYPE_MATCH">VertexAttrib function must match shader attribute type</a></h3>
+
+    <p>
+        If any of the following situations are true, attempting to draw with <code>drawArrays</code>,
+        <code>drawElements</code>, <code>drawRangeElements</code> or their instanced variants generates an
+        <code>INVALID_OPERATION</code> error:
+    </p>
+
+    <ul>
+      <li>
+        <code>vertexAttrib{1234}f</code> or <code>vertexAttrib{1234}fv</code> is used and the base
+        type of the shader at slot <code>index</code> is not floating-point (e.g. is signed or unsigned
+        integer);
+      </li>
+      <li>
+        <code>vertexAttribI4ui</code> or <code>vertexAttribI4uiv</code> is used and the base type of
+        the shader attribute at slot <code>index</code> is floating-point or signed integer;
+      </li>
+      <li>
+        <code>vertexAttribI4i</code> or <code>vertexAttribI4iv</code> is used and the base type of
+        the shader attribute at slot <code>index</code> is floating-point or unsigned integer.
+      </li>
```

Comment:
That is actually unclear from the spec.  Because if we want to distinguish int / uint, we will have to look into the function parameter |type|, not just the function type.  This feels the right thing, only it's not backed up by the ES3 spec. What do you feel?